awscli/customizations/ecs/deploy.py

```diff
@@ -341,6 +342,24 @@ def describe_cd_resources(self):
             raise exceptions.ServiceClientError(
                 action='describe Code Deploy deployment group', error=e)
 
+        try:
+            dgp_info = self.deployment_group_details['deploymentGroupInfo']
+            deployment_config_name = dgp_info['deploymentConfigName']
+            self.deployment_config = self._client.get_deployment_config(
```

Comment:
is `deploymentConfigName` a resource which will always be present in the `get_deployment_group` response, or only when specific types of deployments are configured?

Reply:
`deployemntConfigName` is kind of mandatory. From the API docs:

> If specified, the deployment configuration name can be either one of the predefined configurations provided with AWS CodeDeploy or a custom deployment configuration that you create by calling the create deployment configuration operation.
CodeDeployDefault.OneAtATime is the default deployment configuration. It is used if a configuration isn't specified for the deployment or deployment group.

If no `deployemntConfigName` is provided the system use `CodeDeployDefault.OneAtATime` as default, which leads to an error, because this config is not valid for ECS.

But to be safe, I can for sure here check the existence of the key first and skip the whole rerouting time calculation for this case.

Reply:
Thanks for the context!  Curious, this key isn't currently provided in any of our [functional tests](https://github.com/aws/aws-cli/blob/develop/tests/functional/ecs/test_deploy.py#L223)... though [they're supposed to be run](https://github.com/aws/aws-cli/blob/develop/scripts/ci/run-tests#L19-L20) as an automated check on PRs. If this code is always making a `GetDeplotmentConfig` call with it's value now, how are these passing?? 

Can you try to run the functional test locally and see what happens? Since this call will (nearly) always be made now, I also expect we'll need to update these to include it.